src/impy/common.py

```diff
@@ -610,21 +618,24 @@ def __call__(self, nevents):
         nretries = 0
         for nev in self._composite_plan(nevents):
             while nev > 0:
-                if self._generate_event():
+                if self._generate():
                     nretries = 0
                     self.nevents += 1
                     nev -= 1
-                    yield self._event_class(self)
+                    event = self._event_class(self)
+                    # boost into frame requested by user
+                    self.kinematics.apply_boost(event, self._frame)
+                    yield event
                     continue
                 nretries += 1
-                if nretries > 5:
+                if nretries % 50 == 0:
                     warnings.warn(
-                        "Event was rejected 5 times in a row. The event generator "
+                        f"Event was rejected {nretries} times in a row. The generator "
                         "may be misconfigured or used outside of its valid range",
                         RuntimeWarning,
                     )
-                if nretries > 20:
-                    raise RuntimeError("More than 20 retries, aborting")
+                if nretries > 1000:
```

Comment:
That's really quite high. Which case were you running that required such a high number or retries? Pythia with nuclei?

Reply:
Yes, I was playing with Pythia8 with nuclei. I do not understand what is happening. I got it to work for a while but then it stopped working again, both times consistently over many runs. For now, I would rather have this value large so that we have a chance to get valid Pythia8 events.